addons/l10n_uy/models/l10n_latam_document_type.py

```diff
@@ -0,0 +1,28 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+import re
+
+
+class L10nAccountDocumentType(models.Model):
+
+    _inherit = 'l10n_latam.document.type'
+
+    def _format_document_number(self, document_number):
+        """ format and validate the document_number"""
+        self.ensure_one()
+        if self.country_id.code != "UY":
+            return super()._format_document_number(document_number)
+
+        if not document_number:
+            return
+
+        document_number = document_number.strip()
+        number_part = re.findall(r'[\d]+', document_number)
+        serie_part = re.findall(r'^[A-Za-z]+', document_number)
+
+        if not serie_part or len(serie_part) > 1 or len(serie_part[0]) > 2 \
+           or not number_part or len(number_part) > 1 or len(number_part[0]) > 7:
+            raise UserError(_('Not Valid UY document number (ALPHA 2 - NUM 7)'))
```

Comment:
Hi @zaoral I think this message can be improved to be more user frindly as a regular user friendly,  example:  "Please introduce a valiud Document number:   2 letters and 7 digits  (XX-0000001)"  

We can omit the UY reference as this error will be rise just in this localization.


![image](https://github.com/odoo/odoo/assets/33298180/bf58accc-83dc-45c1-84b8-12d7f15930e9)